pennylane/ops/functions/assert_valid.py

```diff
@@ -187,6 +187,29 @@ def _check_eigendecomposition(op):
         assert qml.math.allclose(decomp_mat, original_mat), failure_comment
 
 
+def _check_generator(op):
+    """Checks that if an operator has a generator, it does."""
+
+    if op.has_generator:
+        gen = op.generator()
+        assert isinstance(gen, qml.operation.Operator)
+        new_op = (
+            qml.exp(gen, -1j * op.data[0])
+            if isinstance(op, qml.ops.Evolution)  # Is this consistency acceptable?
+            else qml.exp(gen, 1j * op.data[0])
+        )
```

Comment:
😢 . Well, at least we are finding this.  I think we need to change `Evolution.generator` to match everywhere else... This is actually worth a bug report on its own.

Reply:
I opened a bug report: https://github.com/PennyLaneAI/pennylane/issues/6311